src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -38,5 +38,47 @@ public static class TaskTimeoutExtensions
                 throw new TimeoutException();
             }
         }
+
+        public static async Task WhenAllOrAnyFailed(this Task[] tasks, int millisecondsTimeout)
+        {
+            var cts = new CancellationTokenSource();
+            Task task = tasks.WhenAllOrAnyFailed();
+            if (task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout, cts.Token)))
+            {
+                cts.Cancel();
+                await task;
```

Comment:
`.ConfigureAwait(false)`... as I noted in the other PR, as this is essentially library code, even though it's purely for tests we do still want to use `.ConfigureAwait(false)`, as otherwise it could mess up tests that, for example, block on the resulting Task.

Reply:
Changed